var/spack/repos/builtin/packages/abacus/package.py

```diff
@@ -6,40 +6,214 @@
+    )
+    version(
+        "2.2.4",
+        sha256="1f3923d32c392fd2564d137b076134aebdc461dc3148c535a00f9b10a615157a",
+        deprecated=True,
+    )
+    version(
+        "2.2.3",
+        sha256="88dbf6a3bdd907df3e097637ec8e51fde13e2f5e0b44f3667443195481320edf",
+        deprecated=True,
+    )
+    version(
+        "2.2.2",
+        sha256="4a7cf2ec6e43dd5c53d5f877a941367074f4714d93c1977a719782957916169e",
+        deprecated=True,
+    )
+    version(
+        "2.2.1",
+        sha256="14feca1d8d1ce025d3f263b85ebfbebc1a1efff704b6490e95b07603c55c1d63",
+        deprecated=True,
+    )
+    version(
+        "2.2.0",
+        sha256="09d4a2508d903121d29813a85791eeb3a905acbe1c5664b8a88903f8eda64b8f",
+        deprecated=True,
+    )
 
+    variant("mpi", default=True, description="Enable MPI support")
     variant("openmp", default=True, description="Enable OpenMP support")
+    variant(
+        "lcao",
+        default=True,
+        description="Enable Linear Combinition of Atomic Orbital calculation",
+        when="+mpi",
+    )
+    variant("libxc", default=True, description="Support additional functionals via libxc")
+    variant(
+        "elpa",
+        default=True,
+        description="Enable optimised diagonalisation routines from ELPA",
+        when="+lcao",
+    )
+    variant("mathlib", default=False, description="Enable ABACUS's builtin libm")
+    variant(
+        "tests", default=False, description="Build ABACUS unit tests", when="build_system=cmake"
+    )
+    variant(
+        "benchmarks",
+        default=False,
+        description="Enable ABACUS's builtin benchmark tests",
+        when="+tests",
+    )
+    variant("rocm", default=False, description="(Experimental)Enable rocm support")
+    # TODO: Add support for
+    # LibRI(https://github.com/abacusmodeling/LibRI),
+    # LibComm(https://github.com/abacusmodeling/LibComm),
+    # Libnpy(https://github.com/llohse/libnpy/),
+    # DeePKS(https://github.com/deepmodeling/deepks-kit),
+    # DeePMD(https://github.com/deepmodeling/deepmd-kit),
+    # LibPAW-interface(https://github.com/wenfei-li/libpaw_interface),
+    # At 2024-1-30, none of above have a spack package.
+
+    depends_on("fftw-api@3")
+    depends_on("blas")
+    depends_on("lapack")
+
+    with when("+mpi"):
+        depends_on("mpi", type=("build", "link", "run"))
+    with when("+libxc"):
+        depends_on("libxc")
+    with when("+lcao"):
+        depends_on("cereal")
+        depends_on("scalapack")
+    with when("+elpa"):
+        depends_on("elpa")
+    with when("+tests"):
+        depends_on("googletest")
+    with when("+benchmarks"):
+        depends_on("benchmark")
+    with when("+openmp"):
+        depends_on("fftw+openmp", when="^[virtuals=fftw-api] fftw")
+        depends_on("elpa+openmp", when="+elpa")
+        depends_on(
+            "openblas threads=openmp",
+            when="^[virtuals=blas] openblas" or "^[virtuals=lapack] openblas",
+        )
```

Comment:
This would evaluate to the first arg, so you can simply use:
```suggestion
        depends_on(
            "openblas threads=openmp",
            when="^[virtuals=blas] openblas",
        )
```
As far as I know `openblas` provides lapack and blas together (it cannot build on top of other blas libraries).

Reply:
In some corner cases, openblas could be concretized as only a lapack provider, e. g.: 
<details><summary>one corner case example</summary>
<p>

yizeyi18@DESKTOP-5UK5DD4:~/soft/spack$ spack spec -l abacus ^netlib-xblas
Input spec
--------------------------------
 -   abacus
 -       ^netlib-xblas

Concretized
--------------------------------
 -   q44hmr5  abacus@develop%gcc@13.2.0~cuda+elpa~ipo+lcao+libxc~mathlib+mpi+openmp~rocm~tests build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-skylake
[+]  uhgc4yl      ^cereal@1.3.2%gcc@13.2.0~ipo build_system=cmake build_type=Release generator=make patches=2dfa0bf arch=linux-ubuntu22.04-skylake
[e]  5jtet3y      ^cmake@3.22.1%gcc@13.2.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-ubuntu22.04-skylake
 -   xccstl6      ^elpa@2023.05.001%gcc@13.2.0~autotune~cuda+mpi+openmp~rocm build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  sjr3v5h          ^libtool@2.4.7%gcc@13.2.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  tdejqfr              ^m4@1.4.19%gcc@13.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu22.04-skylake
[e]  nodstrc                  ^diffutils@3.8%gcc@13.2.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  wq3ghrz                  ^libsigsegv@2.14%gcc@13.2.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[e]  zg3y6s6          ^python@3.10.6%gcc@13.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-ubuntu22.04-skylake
[+]  eonvvfe      ^gcc-runtime@13.2.0%gcc@13.2.0 build_system=generic arch=linux-ubuntu22.04-skylake
[e]  6cnijfi      ^gmake@4.3%gcc@13.2.0~guile build_system=autotools patches=599f134 arch=linux-ubuntu22.04-skylake
[e]  c4hwk6l      ^intel-oneapi-mkl@2023.2.0%gcc@13.2.0~cluster+envmods~ilp64+shared build_system=generic mpi_family=none threads=none arch=linux-ubuntu22.04-skylake
[e]  vkj5jcu      ^intel-oneapi-mpi@2021.10.0%gcc@13.2.0+envmods~external-libfabric~generic-names~ilp64 build_system=generic arch=linux-ubuntu22.04-skylake
[+]  xqbit2c      ^libxc@6.2.2%gcc@13.2.0~cuda~kxc~lxc+shared build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  4kvryat          ^autoconf@2.69%gcc@13.2.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-ubuntu22.04-skylake
[+]  344ajhj          ^automake@1.16.5%gcc@13.2.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  x6tczzn          ^perl@5.38.0%gcc@13.2.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu22.04-skylake
[+]  bbwnq6p              ^berkeley-db@18.1.40%gcc@13.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu22.04-skylake
[+]  aynbov5              ^bzip2@1.0.8%gcc@13.2.0~debug~pic+shared build_system=generic arch=linux-ubuntu22.04-skylake
[+]  2cxlzy6              ^gdbm@1.23%gcc@13.2.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  epwc5ug                  ^readline@8.2%gcc@13.2.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu22.04-skylake
[+]  ghcbbga                      ^ncurses@6.4%gcc@13.2.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  xm23fkm                          ^pkgconf@1.9.5%gcc@13.2.0 build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  saszwn4              ^zlib-ng@2.1.3%gcc@13.2.0+compat+opt build_system=autotools patches=299b958,ae9077a,b692621 arch=linux-ubuntu22.04-skylake
 -   cyx2ffx      ^netlib-scalapack@2.2.0%gcc@13.2.0~ipo~pic+shared build_system=cmake build_type=Release generator=make patches=072b006,1c9ce5f,244a9aa arch=linux-ubuntu22.04-skylake
 -   omuynna      ^netlib-xblas@1.0.248%gcc@13.2.0+fortran+plain_blas build_system=autotools arch=linux-ubuntu22.04-skylake
[+]  wgxa4ex      ^openblas@0.3.23%gcc@13.2.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=openmp arch=linux-ubuntu22.04-skylake

</p>
</details> 
In this example, `netlib-xblas`, `openblas`, `netlib-scalapack` and `intel-oneapi-mkl` serves provider of `blas`, `lapack`, `scalapack` and `fftw-api`.

Reply:
What I mean is:
```
>>> "foo" or "bar"
'foo'
```
so the argument above evaluates as the first operand

Reply:
removed in [f38aa6b](https://github.com/spack/spack/pull/42974/commits/f38aa6b1ff91d5cae79b06ca34314a3e4ee2687b) .